blocks/extensions/training.py

```diff
@@ -0,0 +1,28 @@
+from blocks.extensions import TrainingExtension
+
+
+class AdjustParameter(TrainingExtension):
+    """Adjusts shared variable parameter using some function.
+
+    Applies function to compute the new value of a shared parameter each
+    iteration.
+
+    This class can be used to adapt over the training process parameters like
+    learning rate, momentum, etc.
+
+    Parameters
+    ----------
+    parameter : shared variable to be adjusted
+    function : function which input number of aggregated examples and outputs
+               the parameter value
+
+    """
+    def __init__(self, parameter, function, **kwargs):
+        super(AdjustParameter, self).__init__(**kwargs)
+        self.learning_rate = parameter
+        self.function = function
+        self.num_examples = 0
```

Comment:
There is no need to keep track of this yourself! The total number of examples seen so far is stored in the log. I'm not entirely sure about the precise attribute (@rizar?) but I think it should be something like `self.main_loop.log.current_row.iterations_done`.